server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdSettings.java

```diff
@@ -182,11 +182,13 @@ public void validate(final String value, final Map<Setting<?>, Object> settings)
     }
 
     private static void doValidate(String low, String high, String flood) {
-        try {
-            doValidateAsPercentage(low, high, flood);
-            return; // early return so that we do not try to parse as bytes
-        } catch (final ElasticsearchParseException e) {
-            // swallow as we are now going to try to parse as bytes
+        if (low.endsWith("b") == false) { // only try to validate as percentage if it isn't obviously a byte size value
```

Comment:
Seems a bit odd just to take the shortcut on a lower-case `b` although I see that it helps. Let's extract the predicate to a method `definitelyNotPercentage(String)` so we can add a slightly longer comment about why we do this particular check.